### PR TITLE
Implement dry_run to easily list images to delete

### PR DESCRIPTION
**Pull request by nhuray**
Hi @mirakl-admin, @anouarchattouna, @PascalBourdier and @patricklubach

Thanks for the terraform module to easily cleanup images from Google Cloud Registry, it's really super useful.

I would like to use it but because I'm kind of scare to remove too much images I would like to run `gcr-cleaner` in dry-run mode.

It looks like there's a `dry-run` option  since this commit : https://github.com/GoogleCloudPlatform/gcr-cleaner/commit/1ac1af8320faa110bd20f3555cd8e870ab01a247

So in that PR, I added `dry_run` assuming we are using the latest `gcr-cleaner`image.

Nicolas

### CODE REVIEW

**Comment by anouarchattouna** [comment]
Hi @nhuray, thank you for this PR, we will give it a try shortly!

File: locals.tf [hunk 1]
```diff
@@ -41,6 +41,7 @@ locals {
       tag_filter_any = repo.parameters != null ? (repo.parameters.tag_filter_any != null ? repo.parameters.tag_filter_any : "") : ""
       tag_filter_all = repo.parameters != null ? (repo.parameters.tag_filter_all != null ? repo.parameters.tag_filter_all : "") : ""
       recursive      = true
+      dry_run        = repo.parameters != null ? (repo.parameters.dry_run != null ? repo.parameters.dry_run : false) : false
       filter         = repo.parameters != null ? "grace-${repo.parameters.grace != null ? repo.parameters.grace : "0"}-keep-${repo.parameters.keep != null ? repo.parameters.keep : "0"}-tag_filter-${repo.parameters.tag_filter != null ? repo.parameters.tag_filter : "no"}-tag_filter_any-${repo.parameters.tag_filter_any != null ? repo.parameters.tag_filter_any : "no"}-tag_filter_any-${repo.parameters.tag_filter_any != null ? repo.parameters.tag_filter_any : "no"}" : "delete-all-untagged-images-recursive"
```
Comment (anouarchattouna) [inline locals.tf:45]:
```suggestion
      filter         = repo.parameters != null ? "grace-${repo.parameters.grace != null ? repo.parameters.grace : "0"}-keep-${repo.parameters.keep != null ? repo.parameters.keep : "0"}-tag_filter-${repo.parameters.tag_filter != null ? repo.parameters.tag_filter : "no"}-tag_filter_any-${repo.parameters.tag_filter_any != null ? repo.parameters.tag_filter_any : "no"}-tag_filter_any-${repo.parameters.tag_filter_any != null ? repo.parameters.tag_filter_any : "no"}" : "delete-all-untagged-images-recursive${repo.parameters.dry_run != null ? "${repo.parameters.dry_run ? "-dry_run" : ""}" : ""}"
```

Comment (anouarchattouna) [reply locals.tf:45]: Updating the filter (that will be used in the name of the job) to add `dry-run` if set to true.

File: locals.tf [hunk 2]
```diff
@@ -58,6 +59,7 @@ locals {
           tag_filter_any = repo.tag_filter_any != null ? repo.tag_filter_any : ""
           tag_filter_all = repo.tag_filter_all != null ? repo.tag_filter_all : ""
           recursive      = repo.recursive != null ? repo.recursive : false
+          dry_run        = repo.dry_run != null ? repo.dry_run : false
           filter         = "grace-${repo.grace != null ? repo.grace : "0"}-keep-${repo.keep != null ? repo.keep : "0"}-tag_filter-${repo.tag_filter != null ? repo.tag_filter : "no"}-tag_filter_any-${repo.tag_filter_any != null ? repo.tag_filter_any : "no"}-tag_filter_all-${repo.tag_filter_all != null ? repo.tag_filter_all : "no"}-recursive-${repo.recursive != null ? repo.recursive : false}"
```
Comment (anouarchattouna) [inline locals.tf:63]:
```suggestion
          filter         = "grace-${repo.grace != null ? repo.grace : "0"}-keep-${repo.keep != null ? repo.keep : "0"}-tag_filter-${repo.tag_filter != null ? repo.tag_filter : "no"}-tag_filter_any-${repo.tag_filter_any != null ? repo.tag_filter_any : "no"}-tag_filter_all-${repo.tag_filter_all != null ? repo.tag_filter_all : "no"}-recursive-${repo.recursive != null ? repo.recursive : false}${repo.dry_run != null ? "${repo.dry_run ? "-dry_run" : ""}" : ""}"
```

**Review by anouarchattouna — CHANGES_REQUESTED** [review]
Could you please update the examples accordingly and run `pre-commit run -a` again?

File: locals.tf (hunk 1 above)
Comment (nhuray) [reply locals.tf]: @anouarchattouna I updated the filter with `dry_run` as requested, but I put it only if `parameters` is set

Comment (anouarchattouna) [reply locals.tf]: Thanks!

**Review by lpereir4 — APPROVED** [review]

**Comment by anouarchattouna** [comment]
@nhuray could please run `pre-commit run -a` again and update this PR with any changes?

**Comment by nhuray** [comment]
> pre-commit run -a

@anouarchattouna I did it on local but there's no changes to commit !

**Pull request merged**